Align with skeleton and fix few dev setup issues

Haprog opened the pull request:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/vaadin/vaadin-icons/74)
<!-- Reviewable:end -->

Haprog committed:
fix: 404 error related to removed dependency

Remove the broken reference to "elements-demo-resources/ga.js" which is not used anymore.

Haprog committed:
style: align index.html format with vaadin-element-skeleton [skip ci]

Haprog committed:
fix: generate analysis.json on npm start

Fixes error 'Could not load descriptor "analysis.json".' (404) which caused the whole documentation to be broken (no information) when running "polymer serve".
Also adds "--open" flag to polymer serve to align with vaadin-element-skeleton.

Haprog committed:
chore: align with vaadin-element-skeleton

Add missing editorconfig and lint configs.
Add lint lasks, fix lint issues.
Update vaadin-demo-helpers dev dependency (minor).
Align bower.json, gulpfile.js and package.json.

Haprog committed:
fix: 404 in visual test

Removed unnecessary broken import

Haprog committed:
style: align visual test format with vaadin-element-skeleton

Haprog commented:
I left out the "version:update" related tasks from gulpfile and package.json since this repo doesn't contain custom element definitions and there is no "src" directory so those task aren't relevant here.

Haprog commented:
Linter configs are used mainly for linting the gulpfile, demos and tests in this case.

web-padawan commented on index.html:
```diff
@@ -1,18 +1,16 @@
-  <meta name="viewport" content="width=device-width, initial-scale=1">
-  <title>vaadin-icons</title>
-
-  <script src="../elements-demo-resources/ga.js"></script>
```
this line seems to be the only meaningful change for this file

Haprog replied:
Yes. In addition to removing that unnecessary script I just wanted the code style/formatting of this file to match the skeleton (and other components based on it).

Both of these changes were done in separate commits.

web-padawan commented on gulpfile.js:
```diff
@@ -1,17 +1,68 @@
+
+gulp.task('lint', ['lint:js', 'lint:html', 'lint:css']);
+
+gulp.task('lint:js', function() {
```
I'm currently working on vaadin/vaadin-element-skeleton#199 so the lint tasks would be changed soon. 
Let's exclude this change from the current PR.

Haprog replied:
I understand but we would need to update these in all component repos anyway so I thought it wouldn't matter much even if I change these so that the linting is usable here a bit sooner. Do you expect to complete the skeleton lint task changes soon?

So you want me to remove these lint tasks from gulpfile? I'm fine with that, at least I have linting in VS Code due to the linter configs and deps from `vaadin-component-dev-dependencies`.

Haprog replied:
Lint tasks have now been removed from gulpfile.

web-padawan commented on package.json:
```diff
@@ -16,16 +16,17 @@
   },
-  "homepage": "https://vaadin.com/icons",
   "devDependencies": {
+    "@vaadin/vaadin-component-dev-dependencies": "^1.0.0",
```
After  vaadin/vaadin-element-skeleton#199 is fixed, we will be able to drop gulp from `@vaadin/vaadin-component-dev-dependencies` and only install it where it is actually needed

It makes sense to update gulp to 4.0.0 manually here, though

Haprog replied:
I understand. So do you want any changes here right now? After gulp is removed from `vaadin-component-dev-dependencies` and when we update this here to use the newer version we would need to add gulp dependency here explicitly since it is now needed at least for the icon set generation.

Or do you want me to already add/update the gulp dependency here?

web-padawan reviewed (changes requested):
Let's split the

Haprog committed:
chore: remove lint tasks from gulpfile

Remove lint tasks from gulpfile in preparation for https://github.com/vaadin/vaadin-element-skeleton/issues/199

Haprog reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/vaadin/vaadin-icons/74)* status: 0 of 9 files reviewed, 2 unresolved discussions (waiting on @web-padawan)

___
*[gulpfile.js, line 15 at r1](https://reviewable.io/reviews/vaadin/vaadin-icons/74#-La-V2fmOU_aQUBATF0K-r1-15:-La06owTCrPLppc47uV1:b-896fix) ([raw file](https://github.com/vaadin/vaadin-icons/blob/c151b16cb6a974f2c996c3f3e17ce1c6b0248d20/gulpfile.js#L15)):*
<details><summary><i>Previously, Haprog (Kari Söderholm) wrote…</i></summary><blockquote>

Lint tasks have now been removed from gulpfile.
</blockquote></details>

Done.


<!-- Sent from Reviewable.io -->

web-padawan reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

> Reviewed 8 of 9 files at r1, 1 of 1 files at r2.
> *[Reviewable](https://reviewable.io/reviews/vaadin/vaadin-icons/74#-:-La0A5V60mUTpdcYTCzZ:bnfp4nl)* status: :shipit: complete! all files reviewed, all discussions resolved



<!-- Sent from Reviewable.io -->
